### PR TITLE
Use enums that are more compatible with qt6

### PR DESCRIPTION
**Pull request by hmaarrfk**
xref: https://github.com/napari/napari/pull/5480/files

It seems to be getting removed

See: https://wiki.qt.io/Qt_for_Python_Development_Notes

    old Enums are being removed, and the startup time will improve a bit (still in progress)

# Fixes/Closes

Closes #

# Description

# References

## Type of change

- [ ] Bug-fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How has this been tested?

- [ ] example: the test suite for my feature covers cases x, y, and z
- [ ] example: all tests pass with my change
- [ ] example: I check if my changes works with both PySide and PyQt backends
      as there are small differences between the two Qt bindings.  

## Final checklist:
- [ ] My PR is the minimum possible work for the desired functionality
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] If I included new strings, I have used `trans.` to make them localizable.
      For more information see our [translations guide](https://napari.org/developers/translations.html).

### CODE REVIEW

**Comment by Czaki** [comment]
But here you change to old-style Enums (Qt4 ones).

**Comment by hmaarrfk** [comment]
It seems that this is the style of syntax that they are pushing for now:

https://doc.qt.io/qtforpython-6/considerations.html#the-differences-between-old-and-new-enums

I'm not sure what the `foregiveness mode` is. Should we be enabling it at conda-forge for PySide6 6.5? 

My hunch is that this "new mode" was made default in 6.5 and the `foregiveness mode` was removed:

https://doc.qt.io/qtforpython-6/PySide6/QtCore/QEnum.html

> Meanwhile we have converted all enums and flags to Python Enums (optional in PySide 6.3, default in PySide 6.4), see the [The New Python Enums](https://doc.qt.io/qtforpython-6/considerations.html#newenumsystem) section.

File: napari/_qt/containers/_base_item_view.py [hunk 1]
```diff
@@ -118,7 +118,7 @@ def _sync_selection_models(self):
         for i in self._root.selection:
             idx = index_of(self.model(), i)
             selection.select(idx, idx)
-        sel_model.select(selection, sel_model.SelectionFlag.ClearAndSelect)
+        sel_model.select(selection, QItemSelectionModel.ClearAndSelect)
```
Comment (Czaki) [inline napari/_qt/containers/_base_item_view.py:121]: for example, here old code is equivalent for 

```
QItemSelectionModel.SelectionFlag.ClearAndSelect
```
see doc: https://doc.qt.io/qt-6/qitemselectionmodel.html#SelectionFlag-enum

and your 

```
QItemSelectionModel.ClearAndSelect
```

Is old style enum alias. 

Ad compare it with this section https://doc.qt.io/qtforpython-6/considerations.html#doing-a-smooth-transition-from-the-old-enums from documentation that you linked.

Comment (hmaarrfk) [reply napari/_qt/containers/_base_item_view.py:121]: You are correct. in this case the current version on the `main` branch does work. I was hitting this bug in 0.4.17 (latest conda-forge release) and I didn't realize that this fix is in fact ok.

Apologies for the noise here and thank you for explaining things thoroughly.

**Pull request closed without merging**